### PR TITLE
Add `UD21x18` and `SD21x18` types

### PR DESCRIPTION
**Pull request by andreivladbrg**
Closes https://github.com/PaulRBerg/prb-math/issues/207

I've also added support for `SD21x18`, which is the type for `int128`, its maximum value has the same number of decimals as `uint128`

### CODE REVIEW

**Comment by PaulRBerg** [comment]
Thanks @andreivladbrg. Is this needed for our upcoming releases (V2.1 or V2.2)? Just so I understand how much of a priority this is.

**Comment by andreivladbrg** [comment]
@PaulRBerg just wanted to let you know that i've rebased from main and fixed the conflicts

**Comment by PaulRBerg** [comment]
@andreivladbrg I will review this shortly. In the meantime, it would be helpful if you could write up a changelog and put it in the PR's comment body. This would help me when I release a new version containing the changes in this PR.

**Review by PaulRBerg — APPROVED** [review]
feedback below.

Accepted because I resolved all issues myself.

File: src/sd59x18/Casting.sol [hunk 1]
```diff
@@ -67,12 +68,15 @@ function intoUD2x18(SD59x18 x) pure returns (UD2x18 result) {
 /// @notice Casts an SD59x18 number into UD21x18.
 /// @dev Requirements:
 /// - x must be positive.
+/// - x must be less than or equal to `uMAX_UD21x18`.
```
[the remark this answers is not in the document]
Comment (PaulRBerg) [reply src/sd59x18/Casting.sol:71]: I think you just need to run through the logic of this again.

- yes, `int256` is bigger than `uint128`
- `int256` is converted into `uint128`
- `int256` value has to be checked to not be bigger than `type(uinit128).max`

[the remark this answers is not in the document]
Comment (andreivladbrg) [reply src/sd59x18/Casting.sol:71]: ohh, shhess
my bad

**Pull request merged**